WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -66,51 +66,54 @@ private SystemAwakeChecker(CoinJoinManager coinJoinManager, Func<Task<IPowerSavi
 
 		protected override async Task ActionAsync(CancellationToken cancel)
 		{
-			IPowerSavingInhibitorTask? task = _powerSavingTask;
-
 			switch (CoinJoinManager.HighestCoinJoinClientState)
 			{
 				case CoinJoinClientState.Idle:
-					if (task is not null)
-					{
-						Logger.LogWarning("Computer idle state is allowed again.");
-						await task.StopAsync().ConfigureAwait(false);
-						_powerSavingTask = null;
-					}
+					await ReleaseAllPreventionAsync().ConfigureAwait(false);
+					break;
 
+				case CoinJoinClientState.InProgress or CoinJoinClientState.InCriticalPhase:
+					await PreventSleepAsync().ConfigureAwait(false);
 					break;
 
-				case CoinJoinClientState.InProgress:
-					if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-					{
-						if (task is not null)
-						{
-							if (!task.Prolong(Timeout.Add(TimeSpan.FromMinutes(1))))
-							{
-								Logger.LogTrace("Failed to prolong the power saving task.");
-								task = null;
-							}
-						}
-
-						if (task is null)
-						{
-							Logger.LogTrace("Create new power saving prevention task.");
-							_powerSavingTask = await TaskFactory!().ConfigureAwait(false);
-						}
-					}
-					else
-					{
-						await EnvironmentHelpers.ProlongSystemAwakeAsync().ConfigureAwait(false);
-					}
+				default:
+					throw new ArgumentOutOfRangeException();
+			}
+		}
 
-					break;
+		private async Task PreventSleepAsync()
+		{
+			IPowerSavingInhibitorTask? task = _powerSavingTask;
 
-				case CoinJoinClientState.InCriticalPhase:
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				if (task is not null)
+				{
+					if (!task.Prolong(Timeout))
+					{
+						Logger.LogTrace("Failed to prolong the power saving task.");
+						task = null;
+					}
+				}
 
-					break;
+				if (task is null)
+				{
+					Logger.LogTrace("Create new power saving prevention task.");
+					_powerSavingTask = await TaskFactory!().ConfigureAwait(false);
+				}
+			}
+			else
+			{
+				await EnvironmentHelpers.ProlongSystemAwakeAsync().ConfigureAwait(false);
+			}
+		}
 
-				default:
-					throw new ArgumentOutOfRangeException();
+		private async Task ReleaseAllPreventionAsync()
```

Comment:
The name does not fit too much here, I think. I would call it just:

```suggestion
		private async Task StopTaskAsync()
```